pages/index.js

```diff
@@ -39,10 +57,29 @@ function searchBlogs(posts, keyword) {
   return results.map((result) => result.post);
 }
 
-export default function Home({ posts, status }) {
+export default function Home({ posts, status, categories }) {
+  var [results, setResults] = useState(posts);
+  const [categoryPosts, setCategoryPosts] = useState(posts);
   const [keyword, setKeyword] = useState("");
-  const [results, setResults] = useState(posts);
-  const count = posts.length;
+  const [activeIndex, setActiveIndex] = useState("");
+  const count = results.length;
+  const category = "all";
+
+  const filterBlogs = (event) => {
+    if (event.target.innerHTML == category) {
+      setCategoryPosts(posts);
+      setResults(posts);
+    } else {
+      results = posts.filter(
+        (result) =>
+          result.attributes.category.data != null &&
+          result.attributes.category.data.attributes.name ==
+            event.target.innerHTML
+      );
+      setCategoryPosts(results);
+      setResults(results);
```

Comment:
If we are not using `results` anywhere, lets remove it

Reply:
Actually we used it, because the result variable will display the posts of results, I mean search and filter posts.